### PR TITLE
Create .coderabbit.yaml

### PR DESCRIPTION
**Pull request by aniket866**
###  Screenshots/Recordings/discordmessage:
- as Suggested here  :https://discord.com/channels/1022871757289422898/1235508903450775622/1474646668183081010

## Checklist

- [x] My code follows the project's code style and conventions.
- [x] If applicable, I have made corresponding changes or additions to the documentation.
- [x] If applicable, I have made corresponding changes or additions to tests.
- [x] My changes generate no new warnings or errors.
- [x] I have joined the Stability Nexus's [Discord server]([https://discord.gg/hjUhu33uAn](https://discord.gg/eqYhuFzuKN)) and I will share a link to this PR with the project maintainers there.
- [x] I have read the [Contribution Guidelines](../CONTRIBUTING.md).
- [x] Once I submit my PR, CodeRabbit AI will automatically review it and I will address CodeRabbit's comments.

## AI Usage Disclosure

Check one of the checkboxes below:

- [x] This PR does not contain AI-generated code at all.
- [ ] This PR contains AI-generated code. I have tested the code locally and I am responsible for it.

I have used the following AI models and tools: TODO

### ⚠️ AI Notice - Important!

We encourage contributors to use AI tools responsibly when creating Pull Requests. While AI can be a valuable aid, it is essential to ensure that your contributions meet the task requirements, build successfully, include relevant tests, and pass all linters. Submissions that do not meet these standards may be closed without warning to maintain the quality and integrity of the project. Please take the time to understand the changes you are proposing and their impact.

### CODE REVIEW

File: .coderabbit.yaml [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+      enabled: true
+    eslint:
+      enabled: true
+
+  # Apply the following labels to PRs
+  labeling_instructions:
+    - label: Python Lang
+      instructions: Apply when the PR/MR contains changes to python source-code
+    - label: Solidity Lang
+      instructions: Apply when the PR/MR contains changes to solidity source-code
```
Comment (rohan-pandeyy) [inline .coderabbit.yaml:130]: Unnecessary labeling instructions for Solidity Lang.

File: .coderabbit.yaml [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+    - label: Typescript Lang
+      instructions: Apply when the PR/MR contains changes to javascript or typescript source-code
+    - label: Ergoscript Lang
+      instructions: Apply when the PR/MR contains changes to ergoscript source-code
```
Comment (rohan-pandeyy) [inline .coderabbit.yaml:134]: Unnecessary labeling instructions for Ergoscript Lang.

File: .coderabbit.yaml [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+        - Flag major feature additions or architectural changes that were implemented without prior design discussion (if applicable).
+        - Flag pull requests that mix unrelated changes or multiple concerns in a single submission.
+        - Ensure security-sensitive logic changes are not introduced without adequate test coverage.
+        - Review for common smart contract vulnerabilities, including but not limited to:
+            - Reentrancy
+            - Improper input validation
+            - Access control bypass
+            - Integer overflows/underflows (if using unchecked blocks)
+            - Front-running risks where applicable
+
```
Comment (rohan-pandeyy) [inline .coderabbit.yaml:246]: Asking the AI to check files and other Ethereum-specific vulnerabilities which do not/will never exist in the codebase,

File: .coderabbit.yaml [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+      instructions: |
+        Review test files for:
+        - Comprehensive coverage of contract behavior.
+        - Coverage of success paths, edge cases, and failure/revert scenarios.
+        - Proper validation of access control restrictions.
+        - Verification of event emissions where applicable.
+        - Explicit validation of state changes after each relevant function call.
+        - Adequate test updates whenever contract logic is modified.
+        - Deterministic behavior (tests should not rely on implicit execution order or shared mutable state).
+        - Clear and descriptive test names that reflect the intended behavior being validated.
```
Comment (rohan-pandeyy) [inline .coderabbit.yaml:270]: Solidity test files path instructions should also be removed.

File: .coderabbit.yaml [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+      # Disable docstring coverage checks (let's assume we don't need them)
+      mode: off
+
+  # Exclude these paths from reviews (build artifacts and dependencies)
+  path_filters:
+    - "!**/node_modules/**" # npm dependencies
+    - "!**/android/**" # Native Android build files
+    - "!**/ios/**" # Native iOS build files
+    - "!**/.expo/**" # Expo build cache
+    - "!**/.expo-shared/**" # Expo shared config
```
Comment (rohan-pandeyy) [inline .coderabbit.yaml:85]: Unnecessary clutter.

File: .coderabbit.yaml [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+        - The code adheres to best practices associated with React
+        - The code adheres to best practices associated with React PWA
+        - The code adheres to best practices associated with SPA
+        - The code adheres to best practices recommended by lighthouse or similar tools for performance
+        - The code adheres to best practices associated with Node.js
+        - The code adheres to best practices recommended for performance
+
+        Security:
+        - No exposed API keys or sensitive data
+        - Use expo-secure-store for sensitive storage
```
Comment (rohan-pandeyy) [inline .coderabbit.yaml:190]: Will confuse the AI's context.

File: .coderabbit.yaml [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+            - Integer overflows/underflows (if using unchecked blocks)
+            - Front-running risks where applicable
+
+
+    # Javascript/Typescript test files
+    - path: "**/*.test.{ts,tsx,js,jsx}"
+      instructions: |
+        Review test files for:
+        - Comprehensive coverage of component behavior
+        - Proper use of @testing-library/react-native
```
Comment (rohan-pandeyy) [inline .coderabbit.yaml:253]: Irrelevant for PictoPy

File: .coderabbit.yaml [hunk 8]
```diff
@@ -0,0 +1,280 @@
...
+
+  # Custom review instructions for specific file patterns
+  path_instructions:
+    # TypeScript/JavaScript files
+    - path: "**/*.{ts,tsx,js,jsx}"
+      instructions: |
+        NextJS:
+        - Ensure that "use client" is being used
+        - Ensure that only features that allow pure client-side rendering are used
+        - NextJS best practices (including file structure, API routes, and static generation methods) are used.
```
Comment (rohan-pandeyy) [inline .coderabbit.yaml:175]: Unrelated prompt to PictoPy's Vite based frontend. This might cause the coderabbit to leave confusing and incorrect review comments on our Vite components.

File: .coderabbit.yaml [hunk 9]
```diff
@@ -0,0 +1,280 @@
...
+    yamllint:
+      enabled: true
+    gitleaks:
+      enabled: true
+    checkov:
+      enabled: true
+    detekt:
+      enabled: true
+    eslint:
+      enabled: true
```
Comment (rohan-pandeyy) [inline .coderabbit.yaml:123]: A few of the language linters have unnecessarily been added. Linters that don't even exist in PictoPy. This will increase review time.

Some of these are:
- `swiftlint`: Swift
- `phpstan`: PHP
- `golangci-lint`: Go
- `detekt`: Kotlin

Comment (rohan-pandeyy) [reply .coderabbit.yaml:123]: The file is also missing the actual context of the actual stack PictoPy uses. There are absolutely no .rs path instructions or Rust linters enabled. Consider adding those instead.

**Review by rohan-pandeyy — CHANGES_REQUESTED** [review]
Hi @aniket866, seems like you've entirely copy-pasted the boilerplate config file and which is extensively unrelated to the PictoPy tech stack. The file needs to be updated according to PictoPy's needs and I request you to remove the Blockchain, Next and Mobile/Expo clutter from the file since this might cause coderabbit to hallucinate during the review.
